engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -487,6 +489,14 @@ struct shield_of_the_righteous_t : public holy_power_consumer_t<paladin_melee_at
     {
       p() -> trigger_memory_of_lucid_dreams( 1.0 );
     }
+
+    if ( p() -> conduit.resolute_defender -> ok() && p() -> buffs.ardent_defender -> up() )
+    {
+      p() -> buffs.ardent_defender -> extend_duration( p(),
+        p() -> conduit.resolute_defender.percent() * ad_duration
```

Comment:
You should be able to call p() -> buffs.ardent_defender -> buff_duration() in runtime instead of re-storing the data's duration value in sotr.